server/ingest.go

```diff
@@ -0,0 +1,99 @@
+package server
+
+import (
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/frain-dev/convoy"
+	"github.com/frain-dev/convoy/config"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/pkg/verifier"
+	"github.com/frain-dev/convoy/queue"
+	"github.com/go-chi/render"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func (a *applicationHandler) IngestEvent(w http.ResponseWriter, r *http.Request) {
+	// 1. Retrieve mask ID
+	maskID := r.URL.Query().Get("maskID")
+
+	// 2. Retrieve source using mask ID.
+	source, err := a.sourceRepo.FindSourceByMaskID(r.Context(), maskID)
+	if err != nil {
+		_ = render.Render(w, r, newErrorResponse(err.Error(), http.StatusBadRequest))
+		return
+	}
+
+	// 3. Select verifier based of source config.
+	var v verifier.Verifier
+	verifierConfig := source.Verifier
```

Comment:
do we need to assert that the `verifier` field on the source is not `nil` before proceeding with the verification?